[Keycloak CI] - Quarkus UT (windows-latest) - Keycloak Quarkus Server Deployment

### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

ci

### Describe the bug


> Error:  Errors: 
> Error:    KeycloakMetricsConfigurationTest � Runtime java.lang.RuntimeException: Failed to start quarkus
> Error:    KeycloakNegativeHealthCheckTest � Runtime java.lang.RuntimeException: Failed to start quarkus
> Error:    KeycloakPathConfigurationTest � Runtime java.lang.RuntimeException: Failed to start quarkus
> Error:    KeycloakReadyHealthCheckTest � Runtime java.lang.RuntimeException: Failed to start quarkus
> [INFO] 
> Error:  Tests run: 4, Failures: 0, Errors: 4, Skipped: 0
> 

`Caused by: java.lang.IllegalStateException: fv-az653-80-49239: failed to find an available port in ports [57800, 57801, 57802, 57803]`

https://github.com/keycloak/keycloak/actions/runs/12049067476/job/33595264630

### Version

latest

### Regression

- [ ] The issue is a regression

### Expected behavior

-

### Actual behavior

-

### How to Reproduce?

-

### Anything else?

_No response_

--- a/quarkus/deployment/src/test/java/test/org/keycloak/quarkus/services/health/KeycloakMetricsConfigurationTest.java
+++ b/quarkus/deployment/src/test/java/test/org/keycloak/quarkus/services/health/KeycloakMetricsConfigurationTest.java
@@ -29,8 +29,9 @@ import static io.restassured.RestAssured.given;
 class KeycloakMetricsConfigurationTest {
 
     @BeforeEach
-    void setUpPort() {
+    void setUp() {
         RestAssured.port = 9001;
+        System.setProperty("KC_CACHE", "local"); // avoid flaky port conflicts
     }
 
     @RegisterExtension

--- a/quarkus/deployment/src/test/java/test/org/keycloak/quarkus/services/health/KeycloakMetricsConfigurationTest.java
+++ b/quarkus/deployment/src/test/java/test/org/keycloak/quarkus/services/health/KeycloakMetricsConfigurationTest.java
@@ -20,7 +20,9 @@ import io.quarkus.test.QuarkusUnitTest;
 import io.restassured.RestAssured;
 import org.jboss.shrinkwrap.api.ShrinkWrap;
 import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.AfterAll;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.RegisterExtension;
 
@@ -28,10 +30,19 @@ import static io.restassured.RestAssured.given;
 
 class KeycloakMetricsConfigurationTest {
 
+    @BeforeAll
+    static void setUpAll() {
+        System.setProperty("KC_CACHE", "local");
+    }
+
+    @AfterAll
+    static void tearDownAll() {
+        System.clearProperty("KC_CACHE");
+    }
+
     @BeforeEach
     void setUp() {
         RestAssured.port = 9001;
-        System.setProperty("KC_CACHE", "local"); // avoid flaky port conflicts
     }
 
     @RegisterExtension

--- a/quarkus/deployment/src/test/java/test/org/keycloak/quarkus/services/health/KeycloakNegativeHealthCheckTest.java
+++ b/quarkus/deployment/src/test/java/test/org/keycloak/quarkus/services/health/KeycloakNegativeHealthCheckTest.java
@@ -45,6 +45,7 @@ public class KeycloakNegativeHealthCheckTest {
         agroalDataSource.close();
 
         RestAssured.port = 9001;
+        System.setProperty("KC_CACHE", "local"); // avoid flaky port conflicts
         given()
                 .when().get("/health/ready")
                 .then()

--- a/quarkus/deployment/src/test/java/test/org/keycloak/quarkus/services/health/KeycloakNegativeHealthCheckTest.java
+++ b/quarkus/deployment/src/test/java/test/org/keycloak/quarkus/services/health/KeycloakNegativeHealthCheckTest.java
@@ -51,5 +51,6 @@ public class KeycloakNegativeHealthCheckTest {
                 .then()
                 .statusCode(503)
                 .body(Matchers.containsString("DOWN"));
+        System.clearProperty("KC_CACHE");
     }
 }

--- a/quarkus/deployment/src/test/java/test/org/keycloak/quarkus/services/health/KeycloakPathConfigurationTest.java
+++ b/quarkus/deployment/src/test/java/test/org/keycloak/quarkus/services/health/KeycloakPathConfigurationTest.java
@@ -30,8 +30,9 @@ import static org.hamcrest.CoreMatchers.is;
 class KeycloakPathConfigurationTest {
 
     @BeforeEach
-    void setUpPort() {
+    void setUp() {
         RestAssured.port = 9001;
+        System.setProperty("KC_CACHE", "local"); // avoid flaky port conflicts
     }
 
     @RegisterExtension

--- a/quarkus/deployment/src/test/java/test/org/keycloak/quarkus/services/health/KeycloakPathConfigurationTest.java
+++ b/quarkus/deployment/src/test/java/test/org/keycloak/quarkus/services/health/KeycloakPathConfigurationTest.java
@@ -20,6 +20,8 @@ import io.quarkus.test.QuarkusUnitTest;
 import io.restassured.RestAssured;
 import org.jboss.shrinkwrap.api.ShrinkWrap;
 import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.RegisterExtension;
@@ -29,10 +31,19 @@ import static org.hamcrest.CoreMatchers.is;
 
 class KeycloakPathConfigurationTest {
 
+    @BeforeAll
+    static void setUpAll() {
+        System.setProperty("KC_CACHE", "local");
+    }
+
+    @AfterAll
+    static void tearDownAll() {
+        System.clearProperty("KC_CACHE");
+    }
+
     @BeforeEach
     void setUp() {
         RestAssured.port = 9001;
-        System.setProperty("KC_CACHE", "local"); // avoid flaky port conflicts
     }
 
     @RegisterExtension

--- a/quarkus/deployment/src/test/java/test/org/keycloak/quarkus/services/health/KeycloakReadyHealthCheckTest.java
+++ b/quarkus/deployment/src/test/java/test/org/keycloak/quarkus/services/health/KeycloakReadyHealthCheckTest.java
@@ -30,8 +30,9 @@ import static io.restassured.RestAssured.given;
 public class KeycloakReadyHealthCheckTest {
 
     @BeforeEach
-    void setUpPort() {
+    void setUp() {
         RestAssured.port = 9001;
+        System.setProperty("KC_CACHE", "local"); // avoid flaky port conflicts
     }
 
     @RegisterExtension

--- a/quarkus/deployment/src/test/java/test/org/keycloak/quarkus/services/health/KeycloakReadyHealthCheckTest.java
+++ b/quarkus/deployment/src/test/java/test/org/keycloak/quarkus/services/health/KeycloakReadyHealthCheckTest.java
@@ -21,6 +21,8 @@ import io.restassured.RestAssured;
 import org.hamcrest.Matchers;
 import org.jboss.shrinkwrap.api.ShrinkWrap;
 import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.RegisterExtension;
@@ -29,10 +31,19 @@ import static io.restassured.RestAssured.given;
 
 public class KeycloakReadyHealthCheckTest {
 
+    @BeforeAll
+    static void setUpAll() {
+        System.setProperty("KC_CACHE", "local");
+    }
+
+    @AfterAll
+    static void tearDownAll() {
+        System.clearProperty("KC_CACHE");
+    }
+
     @BeforeEach
     void setUp() {
         RestAssured.port = 9001;
-        System.setProperty("KC_CACHE", "local"); // avoid flaky port conflicts
     }
 
     @RegisterExtension

